src/services/submission_processor.py

```diff
@@ -14,6 +14,17 @@
 log = logging.getLogger(__name__)
 
 
+def validate_file_processable(file: UploadFile) -> None:
+    extension = file.filename.split(".")[-1].lower()
+    if file.content_type != settings.submission_file_type or extension != settings.submission_file_extension:
+        raise HTTPException(status_code=HTTPStatus.UNSUPPORTED_MEDIA_TYPE, detail="Only CSV supported")
```

Comment:
Adding the {file.content_type} and {extension} in the Exception, as well as the failed size below, might be good.  I like adding that stuff not only for the user to see what failed (you know how users are), but also when looking at logs it's clear why.

Otherwise looks good.

Reply:
yeap, added all the details